### ISSUE TITLE
Some deviantart images mistakenly choosing preview image

### ISSUE THREAD

**Issue by lllusion3469** [r888888888/danbooru#3994]
I managed to replicate it somehow here: [example](https://sta.sh/0sniswucvay)

It is downloadable with resolution `1800x940`, as da itself confirms, but when trying to upload to danbooru, the uploaded version is the preview with resolution `1280x669`.

This happens because of this check failing: https://github.com/r888888888/danbooru/blob/5cf6a4391800815d867707b39c1bbf9553efa13e/app/logical/sources/strategies/deviant_art.rb#L73

```
[8] pry(main)> s=Sources::Strategies.find("https://sta.sh/0sniswucvay")
=> #<Sources::Strategies::Stash:0x0570bdc0
 @parsed_referer=nil,
 @parsed_url=#<Addressable::URI:0x2b85cd0 URI:https://sta.sh/0sniswucvay>,
 @parsed_urls=[#<Addressable::URI:0x2b85cd0 URI:https://sta.sh/0sniswucvay>],
 @referer_url=nil,
 @url="https://sta.sh/0sniswucvay",
 @urls=["https://sta.sh/0sniswucvay"]>
[9] pry(main)> [s.api_deviation[:download_filesize], s.api_deviation.dig(:content, :filesize)]
=> [31499, 31499]
[13] pry(main)> [s.api_deviation.dig(:content, :width), s.api_deviation.dig(:content, :height)]
=> [1280, 669]
[15] pry(main)> [s.api_download[:width], s.api_download[:height]]
=> [1800, 940]
```
Even though the preview file is clearly smaller, it also claims to have a file size of 31499B.

Even though this appears to be a bug on deviantart's end, who know whether it's actually going to get fixed.

Is there a reason, why this check needs to be there in the first place?
Why can't it just choose the download file in general whenever available?

**Comment by BrokenEagle** [comment]
I've experienced the same as the above. For whatever reason, Danbooru is selecting the intermediary size instead of the full size, so I reported it on Discord.

> BrokenEagle98 said:
>
> I just tried using the Bookmarklet on Deviantart, and instead of retrieving the fullsize image (404KB), it instead retrieved the intermediate image (213 KB).(edited)
> https://www.deviantart.com/daekazu/art/She-Ra-Princes-of-Power-Comparison-762327546
> ^That was the page I tried it on.
> I ended up having to download the picture to disk, and then uploading it from there.
> However, I wonder how many sample images have been uploaded from DeviantArt because of that...? o_O

It was then later confirmed by @nonamethanks.

> nonamethanks said ([forum #152730](https://danbooru.donmai.us/forum_posts/152730)):
> 
> post #3331823 with https://images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com/intermediary/f/3e8d2551-bfac-4c5b-89c0-5c5fdefb8b8b/dcp3pwc-25b2331f-e2ca-49a8-b7db-9729a09c9fe3.jpg
> 
> It seems danbooru is having problems getting the right pictures for deviantart. I think a script should be run to catch the most recent uploads because I've seen several others with the same problem.

**Comment by r888888888** [comment]
@evazion wrote most of this code so he'd know best. I think it's just to avoid the extra api call to get the download data.

**Comment by evazion** [comment]
Yeah, it was just to reduce the number of API calls, since for Deviant Art it takes up to four HTTP requests to get all the information we need.

> Even when the post is not downloadable, sometimes it is apparently possible to get a higher resolution than api_deviation.dig(:content, :src)

Fixed in #3997.
